code/data_structures/tree/tree_comparer/tree_comparer.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ Part of Cosmos by OpenGenus Foundation
+
+ tree comparer synopsis
+
+template<typename _Tp, typename _Comp = std::equal_to<_Tp> >
+class TreeComparer
+{
+public:
+    using CNodeType = TreeNode<_Tp> const;
+    using CPCNodeType = std::shared_ptr<CNodeType> const;
+
+    bool isSameTree(CPCNodeType p, CPCNodeType q) const;
+
+private:
+    _Comp comp_;
+};
+ */
+
+#include "../tree_node/tree_node.cpp"
+#include <stack>
+#include <functional>
+
+#ifndef TREE_COMPARER
+#define TREE_COMPARER
+
+template<typename _Tp, typename _Comp = std::equal_to<_Tp> >
+class TreeComparer
+{
+public:
+    using CNodeType = TreeNode<_Tp> const;
+    using CPCNodeType = std::shared_ptr<CNodeType> const;
+
+    bool isSameTree(CPCNodeType p, CPCNodeType q) const {
+        std::stack<CPCNodeType> stp, stq;
```

Comment:
These variable names are terrible! stp and stq could easily get confused for one another and switched in the same way!